addons/mail/static/src/chatter/web/form_renderer.scss

```diff
@@ -27,7 +27,7 @@
 }
 
 // Reduce horizontal spacing if the next sibling is an aside chatter
-.o_form_sheet_bg:has(+ .o-mail-Form-chatter.o-aside) {
+.o_content > .o_form_renderer > .o_form_sheet_bg:has(+ .o-mail-Form-chatter.o-aside) {
```

Comment:
This rule becomes far too specific and is unclear that it attempts to not be considered by modals.

By very specific, I mean that `>` operator can easily break by slightly tweaking the template (e.g. add another container, or remove a container, etc.).

What I suggest instead is to not have `.o-mail-Form-chatter` altogether when in a dialog, so that there's no chatter node therefore no problematic `d-none` as the node doesn't exist at all.

In `@mail/form_compiler`, the `compileChatter()` method could have`t-if="__comp__.env.inDialog"` so that it doesn't render a chatter when in a dialog.

This file has a TODO to filter out dialog that was instead made possible with `d-none` is CSS, but the proper way would be to not render at all in compiler. The scrollbar issue of this PR incites for the proper fix I suggest here.